content/en/blog/_posts/2022-12-29-scalable-job-tracking-ga/index.md

```diff
@@ -0,0 +1,148 @@
+have worked on this foundational improvement since Kubernetes 1.22. After
+multiple iterations and scale verifications, this is now the default
+implementation of the Job controller.
+
+Paired with the Indexed [completion mode](/docs/concepts/workloads/controllers/job/#completion-mode),
+the Job controller can handle massively parallel batch Jobs, supporting up to
+100k concurrent Pods.
+
+The new implementation also made possible the development of [Pod failure policy](/docs/concepts/workloads/controllers/job/#pod-failure-policy),
+which is in beta in the 1.26 release.
+
+## How do I use this feature?
+
+To use Job tracking with finalizers, upgrade to Kubernetes 1.25 or newer and
+create new Jobs. You can also use this feature in v1.23 and v1.24, if you have the
+ability to enable the `JobTrackingWithFinalizers` [feature gate](/docs/reference/command-line-tools-reference/feature-gates/)).
+
+If your cluster runs Kubernetes 1.26, Job tracking with finalizers is a stable
+feature. For v1.25, it's behind that feature gate, and your cluster administrators may have
+explicitly disabled it - for example, if you have a policy of not using
+beta features.
+
+Jobs created before the upgrade will still be tracked using the legacy behavior.
+This is to avoid retroactively adding finalizers to running Pods, which might
+introduce race conditions.
+
+For the maximum performance on large Jobs, the Kubernetes project recommends
+using the [Indexed completion mode](/docs/concepts/workloads/controllers/job/#completion-mode).
+In this mode, the control plane is able to track Job progress with less API
+calls.
+
+If you are a developer of operator(s) for batch, [HPC](https://en.wikipedia.org/wiki/High-performance_computing),
+(AI)[https://en.wikipedia.org/wiki/Artificial_intelligence], [ML](https://en.wikipedia.org/wiki/Machine_learning)
+or related workloads, we encourage you to use the Job API to delegate accurate
+progress tracking to Kubernetes. If there is something missing in the Job API
+that forces you to manage plain Pods, the [Working Group Batch](https://github.com/kubernetes/community/tree/master/wg-batch)
+would welcome your feedback and contributions.
+
+### Deprecation notices
+
+During the development of the feature, the control plane added the annotation
+`batch.kubernetes.io/job-tracking` to Jobs created when the feature is enabled.
+This allowed a safe transition for older Jobs, but it was never meant to stay.
+
+In the 1.26 release, we deprecated the annotation `batch.kubernetes.io/job-tracking`
+and the control plane will stop adding it in Kubernetes 1.27.
+Along with that change, we will remove the legacy Job tracking implementation.
+As a result, the job controller will track all Jobs using finalizers and it will
+ignore Pods that don't have a the aforementioned finalizer.
+
+Before you upgrade your cluster to 1.27, we recommend that you verify that there
+are no running Jobs that don't have the annotation. Otherwise, you might observe
```

Comment:
Since the annotation is deprecated, I assume "don't" is unnecessary in this sentence.
```suggestion
are no running Jobs that have the annotation. Otherwise, you might observe
```

Reply:
The original wording is right, I think. The idea is to **ensure** that no Jobs have the deprecated annotation.

Reply:
To me, the original reads similar to this (in my head)-  There shouldn't be a running job observed that doesn't have the annotation. 

With my suggestion (in my head, again) - There shouldn't be a running job observed that has the annotation.

I think it is probably because I'm not a native English speaker.

Reply:
I added a couple of sentences for extra context.